shared/src/main/java/com/couchbase/lite/AbstractReplicator.java

```diff
@@ -563,6 +663,13 @@ private void documentError(boolean pushing, String docID, C4Error error, boolean
             Log.i(TAG, "C4ReplicatorListener.documentError() pushing -> %s, docID -> %s, error -> %s, trans -> %s", pushing ? "true" : false, docID, error, trans ? "true" : false);
             // TODO: Should pass error along to listener
         }
+
+        if(error.getDomain()==0 && error.getCode()==0){
+            //DocumentReplicatedStatus status = new DocumentReplicatedStatus(true, pushing, docID);
+            c4ReplListener.documentEnded(c4repl, pushing, docID,
```

Comment:
This seems to be incorrect. Instead we need to notify the listener added to the replicator.